File: FlyingFish2/app/src/main/java/app/advance/hcmut/cse/flyingfish/SplashActivity.java
package app.advance.hcmut.cse.flyingfish;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;

public class SplashActivity extends AppCompatActivity {
    private ImageButton startButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        startButton = findViewById(R.id.btnStart);
        startButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startProgress(view);
            }
        });
    }

    public void startProgress(View view) {
        Thread thread = new Thread() {
            @Override
            public void run() {
                try {
                    sleep(2000);
                } catch (Exception e) {
                    e.printStackTrace();
                } finally {
                    Intent mainIntent = new Intent(SplashActivity.this, MainActivity.class);
                    startActivity(mainIntent);
                }
            }
        };

        thread.start();
    }

    @Override
    protected void onPause() {
        super.onPause();
        finish();
    }
}

File: STRINGapp/app/src/main/java/com/example/stringapp/Activity/SplashActivity.kt
package com.example.stringapp.Activity

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.example.stringapp.Fragment.ViewPagerManager
import com.example.stringapp.R
import com.example.stringapp.StaticVariable

class SplashActivity: AppCompatActivity() {
    private val sharedPref = StaticVariable.SHARED_PREF
    private val accessToken = StaticVariable.ACCESS_TOKEN

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.splash_activity)

        startProgress()
    }

    private fun startProgress() {
        val thread: Thread = object : Thread() {
            override fun run() {
                try {
                    sleep(1000)
                } catch (e: Exception) {
                    e.printStackTrace()
                } finally {
                    val sharedPreferences: SharedPreferences = getSharedPreferences(sharedPref, MODE_PRIVATE)
                    sharedPreferences.getString(accessToken, "")?.let { Log.d("token", it) }
                    if (sharedPreferences.getString(accessToken, "") != null){
                        val mainIntent = Intent(this@SplashActivity, SelectInterestActivity::class.java)
                        startActivity(mainIntent)
                    } else {
                        val mainIntent = Intent(this@SplashActivity, ViewPagerManager::class.java)
                        startActivity(mainIntent)
                    }
                }
            }
        }
        thread.start()
    }

    override fun onPause() {
        super.onPause()
        finish()
    }
}